onnx/helper.py

```diff
@@ -31,8 +31,10 @@ def make_graph(nodes, name, inputs, outputs, initializer=[]):
     graph = GraphProto()
     graph.node.extend(nodes)
     graph.name = name
-    graph.input.extend(inputs)
-    graph.output.extend(outputs)
+    for x in inputs:
```

Comment:
Since we have decided to make `type` and `shape` as required in the graph input and output, the argument `inputs` and `outputs` in this `make_graph` function should not be a list of string anymore, otherwise, this `make_graph` will actually produce an invalid graph.

Reply:
Good point. Any proposal how we should change it?

Reply:
inputs and outputs should be a list of corresponding proto structs (like initializer), so the original code is fine (you just need to revert the changes in this file).

Reply:
OK. I will add a MakeType() method to make it easier to construct.

Reply:
@bddppq Thanks for fixing my bug. I was going to look at after dinner, but pleasantly found it fixed.